src/main/java/com/mercuryCyclists/Sales/service/InStoreSaleService.java

```diff
@@ -96,68 +97,87 @@ public ResponseEntity<String> registerInStoreSale(InStoreSale inStoreSale, Long
             throw new IllegalStateException(String.format("Store with Id %s does not exist", storeId));
         }
 
-        // Check product exists
-        JsonObject product = saleService.getSaleProduct(inStoreSale);
-        if(product.get("id") == null){
-            throw new IllegalArgumentException(String.format("Invalid product, %s", product));
-        }
-
-        // If there is enough of the product is stock
-        Long productQuantity = product.get("quantity").getAsLong();
-        Long saleQuantity = inStoreSale.getQuantity();
-        if (productQuantity >= saleQuantity) {
+//        // Check product exists
+//        JsonObject product = saleService.getSaleProduct(inStoreSale);
+//        if(product.get("id") == null){
+//            throw new IllegalArgumentException(String.format("Invalid product, %s", product));
+//        }
+//
+//        // If there is enough of the product is stock
+//        Long productQuantity = product.get("quantity").getAsLong();
+//        Long saleQuantity = inStoreSale.getQuantity();
+
+        JsonObject product = saleService.getSaleProductWithQuantity(inStoreSale);
+        if (product != null) {
             // Update and save product
-            product.addProperty("quantity", (productQuantity - saleQuantity));
+//            product.addProperty("quantity", (productQuantity - saleQuantity));
+            product.addProperty("quantity", (product.get("quantity").getAsLong() - inStoreSale.getQuantity()));
             saleService.updateProduct(product);
```

Comment:
Im thinking that this should probably be done in the inventory service, we could name it something like processSale.
if that then returns true we could create the sale from the sale service?